Feature/toolkit v3

k-kondratowicz opened the pull request.

k-kondratowicz committed:
feat: update playground

k-kondratowicz committed:
chore: update dependencies

k-kondratowicz committed:
feat(split-text): migrate to ovee v3

k-kondratowicz committed:
feat(video-autoplay): update VideoAutoplay component to use the latest version of ovee v3

k-kondratowicz committed:
feat(lottie-player): update LottiePlayer component to use the latest version of  ovee v3

k-kondratowicz committed:
feat(lazy-load): update LazyLoad component to use the latest version of ovee v3

k-kondratowicz committed:
feat(google-map): update GoogleMap component to use ovee v3

k-kondratowicz committed:
feat(base-slider): update BaseSlider component to use ovee v3

k-kondratowicz committed:
chore: remove unused mixins and utils

k-kondratowicz committed:
feat(viewport-units): rename module & update Viewport to use the latest version of ovee v3

k-kondratowicz committed:
feat(in-viewport): update InViewport component to use the latest version of ovee

k-kondratowicz committed:
feat(custom-cursor): update custom-cursor module to use the latest version of ovee

przyb commented on packages/core/composables/in-viewport/useInViewport.ts:
```diff
@@ -0,0 +1,76 @@
+
+		const shouldRemoveClass = options?.shouldRemoveClass ?? true;
+
+		unobserver = observeIntersections(
```
Proponuję zrobić composable useIntersectionObserver, na wzór VueUse

k-kondratowicz replied:
zrobiony, zrobiłem w composablach struktrue folderów jak w vueuse, pewnie dobrze by przemyslec jaką stosujemy, tak zeby wszedzie sie zgadzało, ale ta z vueuse miała dla mnie większy sens, bo omijamy zbędny dodatkowy `index.ts`, ktory tylko exportuje composabla

przyb commented on packages/core/modules/viewport-units/ViewportUnits.ts:
```diff
@@ -0,0 +1,26 @@
+import { defineModule, onInit, ref } from 'ovee.js';
+
+export const ViewportUnits = defineModule(({ app }) => {
+	const vh = ref(0);
+	const vw = ref(0);
+
+	onInit(() => {
+		updateUnits();
+
+		app.$on('resize', updateUnits, { target: window });
+	});
+
+	function updateUnits() {
+		vh.value = document.documentElement.clientHeight / 100;
+		vw.value = window.innerWidth / 100;
+
+		document.documentElement.style.setProperty('--vh', `${vh.value}px`);
+		document.documentElement.style.setProperty('--vw', `${vw.value}px`);
+	}
+
+	return {
+		vh,
+		vw,
+		updateUnits,
+	};
+});
```
Proponuję dodać globalny state dostępny w $app z tymi wartościami i dla uproszczenia dostępu, composable który także je wyeksponuje. Może i trochę edge case, ale dałoby to nam dostęp do tych wartości na poziomie kodu, konsystentnych z CSSem bez zastanawiania się z jakiego property w `window/document` brać wartość.

k-kondratowicz replied:
ogarnięte 👌

przyb commented on packages/core/package.json:
```diff
@@ -26,18 +26,18 @@
   "devDependencies": {
-    "@types/lodash": "^4.14.178",
+    "@types/lodash": "^4.17.7",
     "@types/node": "^17.0.12",
     "gsap": "^3.9.1",
     "lit-html": "^2.3.1",
     "lodash": "^4.17.21",
-    "ovee.js": "~2.2.2",
+    "ovee.js": "^3.0.0-alpha.6",
     "typescript": "^4.5.5"
```
lit-html do wywalenia, lodash i GSAP bump do aktualnej, TS spójnie z Ovee3 czyli na 4.9

k-kondratowicz replied:
a już nawet 5.6.2 na ovee, tutaj też do takiej bumpnąłęm

przyb commented on packages/integrations/base-slider/BaseSlider.ts:
```diff
@@ -1,172 +1,152 @@
 	Autoplay,
+	Controller,
 	EffectFade,
+	FreeMode,
+	Grid,
 	Keyboard,
-	Lazy,
+	Manipulation,
+	Mousewheel,
 	Navigation,
 	Pagination,
-	Swiper,
-	SwiperOptions,
+	Parallax,
+	Thumbs,
 	Virtual,
-} from 'swiper';
-import { LazyOptions, NavigationOptions, PaginationOptions, SwiperEvents } from 'swiper/types';
+	Zoom,
+} from 'swiper/modules';
+import { PaginationOptions } from 'swiper/types/modules/pagination';
```
docelowo chciałbym tego uniknąć, przekazać moduły Swipera w konfiguracji komponentu. No i docelowo refaktor w stronę composabla

k-kondratowicz replied:
wywalone, dodałem info o opcjach do docsów

przyb commented on packages/integrations/split-text/README.md:
````diff
@@ -30,7 +23,19 @@ export default [
+})
+```
+
+The `SplitText` component returns an object containing the `SplitText` instance.
````
```suggestion
The `SplitText` composable returns an object containing the `SplitText` instance.
```

Dodatkowo, dookreślić co jest reaktywne a co nie

k-kondratowicz committed:
refactor(composables): reorganize and rename observeIntersections tool

k-kondratowicz committed:
feat(useMediaQuery): add useMediaQuery composable

k-kondratowicz committed:
chore: up TypeScript & gsap

k-kondratowicz committed:
chore(custom-cursor): remove unused imports

k-kondratowicz committed:
Update packages/integrations/split-text/README.md

Co-authored-by: Jakub Przyborowski <kuba@owlsdepartment.com>

k-kondratowicz committed:
refactor(split-text): improve SplitText composable

k-kondratowicz committed:
refactor(viewport-units): improve ViewportUnits module

- Added reactive viewport height (`vh`) and width (`vw`) units to the `app` object
- Created `useViewportUnits` composable to provide access to `vh` and `vw` units

k-kondratowicz committed:
refactor(base-slider): improve BaseSlider component

k-kondratowicz committed:
docs(viewport-units): updated docs

mateusz-ociepa committed:
feat: navtoggle component migrated to v3

mateusz-ociepa committed:
fix: fixes after code review

mateusz-ociepa committed:
fix: renamed variable

k-kondratowicz committed:
Merge pull request #30 from owlsdepartment/feature/nav-toggle

feat: navtoggle component migrated to v3

k-kondratowicz committed:
docs(nav-toggle): updated README.md

k-kondratowicz committed:
refactor(viewport-units): remove unused import in ViewportUnits.ts

k-kondratowicz committed:
refactor(collapsing-header): update CollapsingHeader to use the latest ver. of ovee

k-kondratowicz committed:
feat(parallax-effect): update component to use the latest version of ovee

k-kondratowicz committed:
feat(parallax-effect): added composable

k-kondratowicz committed:
docs: update docs

k-kondratowicz committed:
chore: rename directories

k-kondratowicz committed:
fix(video-autoplay): update import statement for useInViewport
